src/Symfony/Component/Yaml/Parser.php

```diff
@@ -101,6 +122,16 @@ public function parse($value, $flags = 0)
             mb_internal_encoding('UTF-8');
         }
 
+        Inline::$tags = $this->tags;
+        try {
+            return $this->doParse($value, $flags);
+        } finally {
+            Inline::$tags = array();
```

Comment:
Actually you were right, it was creating a bug when entering a block (because of nested parsers).